src/client/jupyter/languageserver/notebookConverter.ts

```diff
@@ -141,6 +152,10 @@ export class NotebookConverter implements Disposable {
                 }
                 list.push(this.toIncomingDiagnostic(location.uri, d));
             });
+        } else if (this.mapOfConcatDocumentsWithCellUris.has(uri.toString())) {
+            (this.mapOfConcatDocumentsWithCellUris.get(uri.toString()) || [])
```

Comment:
Without this, if a notebook is closed, the diagnostics don't go away.
This is because when a notebook is closed, the wrapper is disposed and we don't have the wrapper document, hence we cannot get the cells to clear the diagnostics.